assets/src/js/cookies.js

```diff
@@ -156,12 +156,20 @@ export const setupCookies = () => {
 
       // Update ad storage and analytics storage if Google Consent Mode is enabled
       if (ENABLE_GOOGLE_CONSENT_MODE) {
-        updateGoogleConsent({
```

Comment:
Don't both code snippets produce the same results? 🤔

Reply:
No, because there's a difference between the value for `analytics_storage` not being set at all or `denied`. Not set means the user didn't make a choice, `denied` means they actively denied.
The old code assumed `denied` even if the user wasn't presented with a choice (ENABLE_ANALYTICAL_COOKIES  ==  false). The revised code will not set it if no choice was made.

Let's take Switzerland as an example. We can load analytics with no active consent, but we might want to give people an opportunity to opt out. So for us, the setup in tag manager would look like this:
- `analytics_storage` == `denied`: don't load analytics
- `analytics_storage` == `granted`: load analytics
- **`analytics_storage`  is not set: load analytics**

That's different from a country that needs active consent, they would not load analytics if `analytics_storage` is not set.

Basically, there are 3 possible values, not 2: `granted`, `denied` and NOT SET.